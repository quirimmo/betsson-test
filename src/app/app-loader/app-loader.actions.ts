import { Injectable } from '@angular/core';
import { Action } from 'redux';
import { NgRedux } from '@angular-redux/store';
import { StoreModel } from '../store/store.model';

export interface AppLoaderAction extends Action {
	isLoading: boolean;
}

@Injectable()
export class AppLoaderActions {
	static readonly START_LOADING = 'START_LOADING';
	static readonly STOP_LOADING = 'STOP_LOADING';

	constructor(private ngRedux: NgRedux<StoreModel>) {}

	public startLoading(): AppLoaderAction {
		return this.ngRedux.dispatch({
			type: AppLoaderActions.START_LOADING,
			isLoading: true
		});
	}

	public stopLoading(): AppLoaderAction {
		return this.ngRedux.dispatch({
			type: AppLoaderActions.STOP_LOADING,
			isLoading: false
		});
	}
}
