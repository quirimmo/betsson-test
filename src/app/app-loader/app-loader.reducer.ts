import { AppLoaderAction, AppLoaderActions } from './app-loader.actions';

export function appLoaderReducer(
	state: boolean = false,
	action: AppLoaderAction
): boolean {
	switch (action.type) {
		case AppLoaderActions.START_LOADING:
			return action.isLoading;
		case AppLoaderActions.STOP_LOADING:
			return action.isLoading;
		default:
			return state;
	}
}
