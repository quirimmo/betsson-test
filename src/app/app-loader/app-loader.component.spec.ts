import { TestBed, async } from '@angular/core/testing';
import { AppLoaderComponent } from './app-loader.component';

const providers = [];
const imports = [];
const declarations = [AppLoaderComponent];

describe('AppLoader Component', () => {
	let componentFixture, componentInstance;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations,
			providers,
			imports
		}).compileComponents();
		componentFixture = TestBed.createComponent(AppLoaderComponent);
		componentInstance = componentFixture.debugElement.componentInstance;
	}));

	it('should create the component', () => {
		expect(componentFixture).toBeTruthy();
	});

	it('should match the snapshot', () => {
		expect(componentFixture).toMatchSnapshot();
	});
});
