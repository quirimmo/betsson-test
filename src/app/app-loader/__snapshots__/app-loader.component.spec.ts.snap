// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader Component should match the snapshot 1`] = `
<app-loader>
  
</app-loader>
`;
